main.py

```
@@ -194,27 +258,55 @@ def __init__(self, root: tk.Tk) -> None:
+        if len(self.trade_history) > self.trade_history_limit:
+            self.trade_history = self.trade_history[-self.trade_history_limit:]
+        self.automation_runner = AutomationRunner(self, self.persistence)
 
         # UI Vars
         self.terminal1_var = tk.StringVar(value=DEFAULT_TERMINAL_1)
         self.terminal2_var = tk.StringVar(value=DEFAULT_TERMINAL_2)
-        self.pair1_var = tk.StringVar()
-        self.lot1_var = tk.StringVar()
-        self.pair2_var = tk.StringVar()
-        self.lot2_var = tk.StringVar()
+        primary_default = self.config.primary_threads[0] if self.config.primary_threads else _default_primary_threads()[0]
```

Ruff (0.13.1)

Undefined name `_default_primary_threads`

(F821)